app/weeklyPicks/page.tsx

```diff
@@ -35,76 +37,75 @@ export default function WeeklyPickForm() {
   const [userPick, setUserPick] = useState<string | null>(null);
   const [userId, setUserId] = useState<string | null>(null);
   const [allPicks, setAllPicks] = useState<string | null>(null);
+  const { isSignedIn } = useAuthContext();
+  const router = useRouter();
 
   useEffect(() => {
-    async function fetchWeeklyPicks() {
-      try {
-        const allPicks = await getAllWeeklyPicks();
-
-        const data: any = await getNFLTeams();
-
-        const response = data.documents;
-        setNFLTeams(response);
-
-        const user = await account.get();
+    if (!isSignedIn) {
+      router.push('/login');
+      return;
+    }
 
-        setUserId(user.$id);
+    const fetchWeeklyPicks = async () => {
+      try {
+        const [allPicksData, nflTeamsData, user] = await Promise.all([
+          getAllWeeklyPicks(),
+          getNFLTeams(),
+          account.get(),
+        ]);
 
         const userPickedTeam = await getUserWeeklyPick({
           userId: user.$id,
           weekNumber: '6622c75658b8df4c4612',
         });
 
-        if (userPickedTeam) {
-          setUserPick(userPickedTeam);
+        if (!nflTeamsData) {
+          console.error('NFL teams data is undefined');
+          setNFLTeams([]);
+        } else {
+          setNFLTeams(nflTeamsData.documents);
         }
 
-        if (allPicks?.documents[0].userResults != '') {
-          setAllPicks(allPicks?.documents[0].userResults);
-        }
+        setUserId(user.$id);
+        setUserPick(userPickedTeam);
+        setAllPicks(allPicksData?.documents[0].userResults || null);
       } catch (error) {
-        console.error(error);
+        console.error('Fetching error:', error);
       }
-    }
+    };
 
     fetchWeeklyPicks();
-  }, []);
+  }, [isSignedIn]);
 
   const form = useForm<z.infer<typeof FormSchema>>({
     resolver: zodResolver(FormSchema),
   });
 
-  async function onSubmit(data: z.infer<typeof FormSchema>) {
-    try {
-      const teamSelect = data.type.toLowerCase();
-
-      if (typeof window !== 'undefined') {
-        window.localStorage.setItem('team', data.type);
-      }
-
-      const findTeamID = NFLTeams.find(
-        (ele) => ele.teamName.toLowerCase() === teamSelect,
-      );
-
-      let appendNewResult;
-
-      const newObj = `"${userId}":{"team":"${findTeamID?.$id}","correct":true}`;
-
-      if (allPicks === null) {
-        appendNewResult = `{${newObj}}`;
-      } else {
-        appendNewResult = `{${allPicks},${newObj}}`;
+  const onSubmit = useCallback(
```

Comment:
nit: this could potentially be it's own function locally elsewhere, and be imported into this file to make the code a bit cleaner.

Reply:
Agreed.  I think we can move this to a server action later on.